src/test/index.ts

```diff
@@ -31,8 +31,91 @@ type TestFixtures = PlaywrightTestArgs & PlaywrightTestOptions & {
+  private _browserContext?: BrowserContext;
+  private _uniqueOrigins = new Set<string>();
+  private _options?: BrowserContextOptions;
+
+  isEnabled(): boolean {
+    return !!process.env.PWTEST_REUSE_CONTEXT;
+  }
+
+  hasContext(): boolean {
+    return !!this._browserContext;
+  }
+
+  setContext(context: BrowserContext) {
+    assert(!this._browserContext);
+    this._browserContext = context;
+    this._options = (context as any)._options;
+    this._browserContext.on('page', page => page.on('framenavigated', frame => this._uniqueOrigins.add(new URL(frame.url()).origin)));
+  }
+
+  async obtainContext(newContextOptions: BrowserContextOptions): Promise<BrowserContext> {
+    assert(this._browserContext);
+    let pages = this._browserContext.pages();
+    const page = pages[0];
+    if (this._uniqueOrigins.size > 0) {
+      let pageWorker = null;
+      if (pages.length > 1) {
+        await page.bringToFront();
+        pageWorker = pages[1];
+      } else {
+        pageWorker = await this._browserContext.newPage();
+        pages = this._browserContext.pages();
+      }
+      await Promise.all(pages.slice(2).map(page => page.close()));
+      await pageWorker.route(url => this._uniqueOrigins.has(url.origin), route => route.fulfill({body: `<html></html>`, contentType: 'text/html'}));
+      for (const origin of this._uniqueOrigins) {
+        await pageWorker.goto(origin);
+        await pageWorker.evaluate(() => window.localStorage.clear());
+        await pageWorker.evaluate(() => window.sessionStorage.clear());
+      }
+      await pageWorker.close();
+    } else {
+      await Promise.all(pages.slice(1).map(page => page.close()));
+    }
+
+    await this._browserContext.clearCookies();
+    await this._applyNewContextOptions(page, newContextOptions);
+    await page.goto('about:blank');
+    this._uniqueOrigins.clear();
+    return this._browserContext;
+  }
+
+  private async _applyNewContextOptions(page: Page, newOptions: BrowserContextOptions) {
+    assert(this._options);
+    if (
+      (
+        this._options.viewport?.width !== newOptions.viewport?.width ||
+        this._options.viewport?.height !== newOptions.viewport?.height
+      ) &&
+      (newOptions.viewport?.height && newOptions.viewport?.width)
+    )
+      await page.setViewportSize({width: newOptions.viewport?.width, height: newOptions.viewport?.height });
+    const emulateMediaOptions: Partial<Parameters<Page['emulateMedia']>[0]> = {};
+    if (this._options.colorScheme !== newOptions.colorScheme)
+      emulateMediaOptions.colorScheme = newOptions.colorScheme;
+    if (this._options.forcedColors !== newOptions.forcedColors)
+      emulateMediaOptions.forcedColors = newOptions.forcedColors;
+    if (this._options.reducedMotion !== newOptions.reducedMotion)
+      emulateMediaOptions.reducedMotion = newOptions.reducedMotion;
+    if (Object.keys(emulateMediaOptions).length > 0)
+      await page.emulateMedia(emulateMediaOptions);
```

Comment:
This is a lot of code for a seemingly random property. Just remove it. Keeping viewport is probably ok.

Reply:
ping?

Reply:
I thought its good to have, not much code and seems to work fine?